lib/keyword-highlighter.js

```diff
@@ -19,17 +19,24 @@ function wrapper (match) {
 }
 
 /**
- * highilight does Simple String Substitution given a set of keywords in
+
+ * highlight does Simple String Substitution given a set of keywords in
  * a body of text and substitutes the keyword with a marked up html Tag
  * @param {String} keywords - a string of one or more keywords
  *   separated by spaces.
  * @param {String} text - the body of text where you want to highlight keywords
- * @returns {String} text - the text with all highilights.
+ * @returns {String} text - the text with all highlights.
  */
 function highlight(keywords, text) {
-  var regex = keywords.split(' ').join('|');
+
+  var regex = keywords.split(' ');
+  regex = regex.filter(function(char){
+    return char !== '';
+  });
+  regex = regex.join('|');
   regex = regex.replace(/[-[\]{}()*+?.,\\^$]/g, "\\$&");
```

Comment:
@SimonLab please add comments to your regex next time it helps others understand the code...